src/scripts/main.js

```diff
@@ -1,3 +1,318 @@
+    moveDown();
+    toDown();
+    moveDown();
+    moveDown();
+  }
+
+  if (button.className.includes('restart')) {
+    appearCell();
+  };
+
+  winMessange();
+  loseMessange();
+});
+
+function toRight() {
+  arrayRows.forEach((el) => {
+    for (let i = 3; i >= 0; i--) {
+      const rowCells = el.children;
+
+      const theCell = rowCells[i];
+      const prevCell = rowCells[i - 1];
+
+      adder(prevCell, theCell);
+    };
+  });
+};
+
+function moveRight() {
+  arrayRows.forEach((el) => {
+    for (let i = 3; i >= 0; i--) {
+      const rowCells = el.children;
+
+      const theCell = rowCells[i];
+      const prevCell = rowCells[i - 1];
+
+      mover(prevCell, theCell);
+    };
+  });
+};
+
+function toLeft() {
+  arrayRows.forEach((el) => {
+    for (let i = 0; i <= 3; i++) {
+      const rowCells = el.children;
+
+      const theCell = rowCells[i];
+      const prevCell = rowCells[i + 1];
+
+      adder(prevCell, theCell);
+    };
+  });
+};
+
+function moveLeft() {
+  arrayRows.forEach((el) => {
+    for (let i = 0; i <= 3; i++) {
+      const rowCells = el.children;
+
+      const theCell = rowCells[i];
+      const prevCell = rowCells[i + 1];
+
+      mover(prevCell, theCell);
+    };
+  });
+};
+
+function createColumns() {
+  const column1 = [];
+  const column2 = [];
+  const column3 = [];
+  const column4 = [];
+
+  arrayRows.forEach((row) => {
+    const rowCells = row.children;
+
+    column1.push(rowCells[0]);
+    column2.push(rowCells[1]);
+    column3.push(rowCells[2]);
+    column4.push(rowCells[3]);
+  });
+
+  const array = [column1, column2, column3, column4];
+
+  return (array);
+};
+
+function moveUp() {
+  columnsArray.forEach((el) => {
+    for (let i = 0; i <= 3; i++) {
+      const theCell = el[i];
+      const prevCell = el[i + 1];
+
+      mover(prevCell, theCell);
+    };
+  });
+};
+
+function toUp() {
+  columnsArray.forEach((el) => {
+    for (let i = 0; i <= 3; i++) {
+      const theCell = el[i];
+      const prevCell = el[i + 1];
+
+      adder(prevCell, theCell);
+    };
+  });
+};
+
+function moveDown() {
+  columnsArray.forEach((el) => {
+    for (let i = 3; i >= 0; i--) {
+      const theCell = el[i];
+      const prevCell = el[i - 1];
+
+      mover(prevCell, theCell);
+    };
+  });
+};
+
+function toDown() {
+  columnsArray.forEach((el) => {
+    for (let i = 3; i >= 0; i--) {
+      const theCell = el[i];
+      const prevCell = el[i - 1];
+
+      adder(prevCell, theCell);
+    };
+  });
+};
+
+function winMessange() {
+  const cellsArray = [...cells];
+
+  for (const cell of cellsArray) {
+    if (cell.textContent === '2048') {
+      messangeWin.classList.remove('hidden');
+    }
+  }
+};
+
+function loseMessange() {
+  const can = canMove();
+
+  if (can === false) {
+    messangeLose.classList.remove('hidden');
+  }
+};
+
+function canMove() {
+  const myArray = [...cells];
+  const freeCells = myArray.filter(cell => cell.textContent === '');
+
+  let can = 'yes';
```

Comment:
```suggestion
  let hasPossibleMoves = true;
```
maybe 'true' ?

Reply:
i tried, but it doesn`t work :( i need 3 options: true, false and YES. True for freeCells and Yes, when i do not have free cells, but have two identical cells in a row.